meta-ostro/classes/supported-recipes.bbclass

```diff
@@ -0,0 +1,288 @@
+                        str(ex)))
+            self.pn_re = parse_regex(parts[0])
+            self.collection_re = parse_regex(parts[1])
+
+        def supported(self, pn, collection):
+            supported = bool((pn is None or self.pn_re[0].match(pn)) and (collection is None or self.collection_re[0].match(collection)))
+            return supported
+
+    class SupportedRecipes:
+        def __init__(self):
+            self.supported = []
+
+        def append(self, recipe):
+            self.supported.append(recipe)
+
+        def current_recipe_supported(self, d):
+            pn = d.getVar('PN', True)
+            filename = d.getVar('FILE', True)
+            collection = bb.utils.get_file_layer(filename, d)
+            return self.recipe_supported(pn, collection)
+
+        def recipe_supported(self, pn, collection):
+            for supported_recipe in self.supported:
+                if supported_recipe.supported(pn, collection):
+                    return True
+            return False
+
+        def find_collections(self, pn):
+            collections = set()
+            for supported_recipe in self.supported:
+                if supported_recipe.supported(pn, None):
+                   collections.add(supported_recipe.collection_re[1])
+            return collections
+
+    files = []
+    supported_files = d.getVar('SUPPORTED_RECIPES', True)
+    if not supported_files:
+        bb.fatal('SUPPORTED_RECIPES is not set')
+    supported_recipes = SupportedRecipes()
+    for file in supported_files.split():
+        try:
+            with open(file) as f:
+                linenumber = 1
+                for line in f.readlines():
+                    if line.startswith('#'):
+                        continue
+                    # TODO (?): sanity check the content to catch obsolete entries or typos.
+                    pn = line.strip()
+                    if pn:
+                        supported_recipes.append(SupportedRecipe(line.strip(), file, linenumber))
+                    linenumber += 1
+            files.append(file)
+        except OSError, ex:
+            bb.fatal('Could not read SUPPORTED_RECIPES = %s: %s' % (supported_file, str(ex)))
+
+    return (supported_recipes, files)
+
+python () {
+    supported_recipes, files = load_supported_recipes(d)
+    # The bitbake cache must be told explicitly that changes in these
+    # files have an effect on the recipe. Otherwise adding
+    # or removing entries does not trigger re-parsing and re-building.
+    for file in files:
+        bb.parse.mark_dependency(d, file)
+    if not supported_recipes.current_recipe_supported(d):
+        d.setVar('EXCLUDE_FROM_WORLD', '1')
+}
+
+python supported_recipes_eventhandler() {
+    supported_recipes, files = load_supported_recipes(d)
+    supported_recipes_check = d.getVar('SUPPORTED_RECIPES_CHECK', True)
+    if not supported_recipes_check:
+        return
+
+    import re
+    isnative_exception = re.compile('|'.join((
```

Comment:
I just worried for scenarios where e.g. glibc would be replaced with musl. or uninative is cripled in.
Having those explicitly declared in visible place might help some developers in future.
E.g. if this list mandatory to OE-core, then put it as OE-core list.
Not critical to address now, just corner case that might surprise people who haven't looked at class implementation.